ecl/hthor/hthor.cpp

```diff
@@ -2647,16 +2714,45 @@ void CHThorHashDedupActivity::stop()
 
 const void * CHThorHashDedupActivity::nextRow()
 {
-    while(true)
+    if (keepBest)
     {
-        OwnedConstRoxieRow next(input->nextRow());
-        if(!next)
+        // Populate hash table with best rows
+        if (!hashTableFilled)
         {
-            table.kill();
-            return NULL;
+            OwnedConstRoxieRow next(input->nextRow());
+            while(next)
+            {
+                table.insertBest(next);
+                next.setown(input->nextRow());
+            }
+            hashTableFilled = true;
+            hashDedupTableIter.first();
+        }
+
+        // Iterate throw hash table returning rows
+        if (hashDedupTableIter.isValid())
+        {
+            HashDedupElement &el = hashDedupTableIter.query();
+
+            OwnedConstRoxieRow row(el.queryRow());
```

Comment:
It doesn't look like the original code coped with hashed grouped.  Hashed dedup is used when the dataset has not been sorted, so does it make sense to have 'unsorted' grouped dataset?  Basically, does it actually make sense to have Hashed dedup with grouped ds?